python/res/fm/ecl/ecl_run.py

```diff
@@ -164,77 +171,77 @@ def initMPI(self):
                 fileH.write("%s\n" % host)
 
 
-
-    def initRedirect(self):
+    def execSerialEclipse(self):
         stdout_file = "%s.LOG" % self.base_name
         stderr_file = "%s.stderr" % self.base_name
+        with open(stdout_file, "w") as stdout, open(stderr_file, "w") as stderr:
+            process = subprocess.Popen(
+                [self.sim.executable, self.base_name],
+                env=self.sim.env,
+                stdout=stdout,
+                stderr=stderr
+            )
+            process.communicate()
+            return process.wait()
 
-        # Redirecting stdout / stderr
-        fd_stdout = os.open(stdout_file , os.O_WRONLY | os.O_TRUNC | os.O_CREAT , 0o644);
-        fd_stderr = os.open(stderr_file , os.O_WRONLY | os.O_TRUNC | os.O_CREAT , 0o644);
 
-        os.dup2(fd_stdout , 1)
-        os.dup2(fd_stderr , 2)
+    def execParallellEclipse(self):
+        self.initMPI( )
+        stdout_file = "%s.LOG" % self.base_name
+        stderr_file = "%s.stderr" % self.base_name
+        with open(stdout_file, "w") as stdout, open(stderr_file, "w") as stderr:
+            process = subprocess.Popen(
+                [self.sim.mpirun,
+                 "-machinefile",
+                 self.machine_file,
+                 "-np",
+                 "%s" % self.num_cpu,
+                 self.sim.executable,
+                 self.base_name],
+                env=self.sim.env,
+                stdout=stdout,
+                stderr=stderr)
+            process.communicate()
+            return process.wait()
 
-        os.close(fd_stdout)
-        os.close(fd_stderr)
 
+    def execEclipse(self):
+        with temp_dir_change():
```

Comment:
Suggestion: How about if we passed the runpath as an argument. Something along the lines of:

```python
with pushd(self.run_path):
    # Do some stuff
```